pre_commit/languages/docker.py

```diff
@@ -37,15 +37,118 @@ def _is_in_docker() -> bool:
         return False
 
 
-def _get_container_id() -> str:
+def _is_in_docker_sched() -> bool:
+    try:
+        with open('/proc/1/sched', 'rb') as f:
+            line = f.readline()
+
+            if line.startswith(b'systemd ') or line.startswith(b'init '):
+                return False
```

Comment:
this isn't a valid detection either -- this just checks what process pid1

Reply:
Well, checking for word `docker` anywhere in the cgroups is also not very exact solution.

Reply:
the scheduling information is just wrong so that one should not be checked

Reply:
So are you in favor of the `.dockerenv` solution?

Reply:
> the scheduling information is just wrong so that one should not be checked

Btw. why do you think it's wrong? I would rather ignore users who run systemd/initd in Docker (it's bad practice anyway) than make `pre-commit` fail on users who run legitimit containers.

Reply:
the identity of pid1 has nothing to do with whether it is running in docker or not

Reply:
> the identity of pid1 has nothing to do with whether it is running in docker or not

The same we could say if there is the word `docker` anywhere in the cgroups.

I still believe that adding more detection methods would make more people happy than unhappy. Or can you think of any other solution that could solve this problem?